engine/access/rpc/backend/backend_stream_blocks.go

```diff
@@ -319,7 +319,7 @@ func (b *backendSubscribeBlocks) getBlock(height uint64, expectedBlockStatus flo
 
 // validateHeight checks if the given block height is valid and available based on the expected block status.
 // Expected errors during normal operation:
-// - storage.ErrNotFound: block for the given block height is not available.
+// - subscription.ErrBlockNotReady when unable to retrieve the block by height.
 func (b *backendSubscribeBlocks) validateHeight(height uint64, expectedBlockStatus flow.BlockStatus) error {
 	highestHeight, err := b.blockTracker.GetHighestHeight(expectedBlockStatus)
```

Comment:
what errors can this return? please double check and add handling to convert to `subscription.ErrBlockNotReady` for any that should be handled

Reply:
Checked and it could return only `codes.InvalidArgument `, so it's not any of the list `errors.Is(err, storage.ErrNotFound) || errors.Is(err, storage.ErrHeightNotIndexed) || execution_data.IsBlobNotFoundError(err)`.